src/utils/getTokenPriceInUSD.ts

```diff
@@ -0,0 +1,175 @@
+  const priceFeed = new ethers.Contract(
+    feedAddress,
+    aggregatorV3InterfaceABI,
+    provider,
+  );
+  // We get the data from the last round of the contract
+  // Determine how many decimals the price feed has (10**decimals)
+  const [roundData, decimals] = await Promise.all([
+    priceFeed.latestRoundData(),
+    priceFeed.decimals(),
+  ]);
+  // We convert the price to a number and return it
+  return Number(
+    (roundData.answer.toString() / Math.pow(10, Number(decimals))).toFixed(2),
+  );
+};
+
+const tokenAddressToFeedAddress = (chainId: ChainId, tokenAddress: string) => {
+  const currencies = currenciesByNetwork[chainId];
+  const currency = Object.values(currencies).find(
+    (currency) => currency.address === tokenAddress,
+  );
+
+  if (!currency) {
+    throw new Error("Currency not supported");
+  }
+
+  const symbol = currency.symbol as keyof (typeof currenciesByNetwork)[ChainId];
+  const feedsForChain = feedsPerChain[chainId];
+
+  if (!feedsForChain) {
+    throw new Error("Chain not supported");
+  }
+
+  return feedsForChain?.[symbol];
+};
+
+export const getTokenPricesForChain = async (chainId: ChainId) => {
+  const currencies = currenciesByNetwork[chainId];
+  const prices = await Promise.all(
+    Object.values(currencies).map(async (currency: Currency) => {
+      try {
+        const price = await getTokenPriceInUSD(chainId, currency.address);
+        return { ...currency, price };
+      } catch (error) {
+        console.error(error);
+        return { ...currency, price: 0 };
+      }
+    }),
+  );
+
+  return prices.reduce(
+    (acc, result) => {
+      return { ...acc, [result.address]: result };
+    },
+    {} as Record<string, Currency & { price: number }>,
+  );
+};
+
+type CurrencyFeeds = Record<
+  keyof (typeof currenciesByNetwork)[ChainId],
+  string
+>;
+
+const feedsPerChain: Record<ChainId, Partial<CurrencyFeeds>> = {
+  [ChainId.SEPOLIA]: {
+    ETH: "0x694AA1769357215DE4FAC081bf1f309aDC325306",
+    WETH: "0x694AA1769357215DE4FAC081bf1f309aDC325306",
+    DAI: "0x14866185B1962B63C3Ea9E03Bc1da838bab34C19",
+    USDC: "0xA2F78ab2355fe2f984D808B5CeE7FD0A93D5270E",
+  },
+  [ChainId.HARDHAT]: {
+    ETH: "0x694AA1769357215DE4FAC081bf1f309aDC325306",
+    WETH: "0x694AA1769357215DE4FAC081bf1f309aDC325306",
+    DAI: "0x14866185B1962B63C3Ea9E03Bc1da838bab34C19",
+    USDC: "0xA2F78ab2355fe2f984D808B5CeE7FD0A93D5270E",
+  },
+  [ChainId.BASE]: {
+    ETH: "0x71041dddad3595F9CEd3DcCFBe3D1F4b0a16Bb70",
+    WETH: "0x71041dddad3595F9CEd3DcCFBe3D1F4b0a16Bb70",
+    DAI: "0x591e79239a7d679378eC8c847e5038150364C78F",
+    USDC: "0x7e860098F58bBFC8648a4311b374B1D669a2bc6B",
+  },
+  [ChainId.OPTIMISM]: {
+    ETH: "0xb7B9A39CC63f856b90B364911CC324dC46aC1770",
+    WETH: "0xb7B9A39CC63f856b90B364911CC324dC46aC1770",
+    DAI: "0x8dBa75e83DA73cc766A7e5a0ee71F656BAb470d6",
+    USDC: "0x16a9FA2FDa030272Ce99B29CF780dFA30361E0f3",
+  },
+  [ChainId.CELO]: {
+    // TODO: DAI on CELO not supported
```

Comment:
Moved it to a json file. I don't see the benefits of leaving out parts of the ABI, in case we need them at some other point in time.

Reply:
Oops, something got lost here. My idea was to default to the USDC price feed as they are both stable coins, so it shouldn't make much of a difference. Defaulting to DAI on mainnet adds a very specific edge case for just one token, because you need to initialize the publicClient to a different chain etc. It's more messy, for little benefit but it's an option too.